File: php_src/bourdon/kernel/app/model/specialisation.php
<?php
require_once(LIB."Model.php");
class specialisation extends Model
{
    protected $id;
    protected $libelle;
    protected $code;

    public function __construct()
    {
        parent::__construct("specialisation", "id");
        $this->id = null;
        $this->libelle = null;
        $this->code = null;
    }

    /**
     * @return null
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param null $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return null
     */
    public function getLibelle()
    {
        return $this->libelle;
    }

    /**
     * @param null $libelle
     */
    public function setLibelle($libelle)
    {
        $this->libelle = $libelle;
    }

    /**
     * @return null
     */
    public function getCode()
    {
        return $this->code;
    }

    /**
     * @param null $libelle
     */
    public function setCode($code)
    {
        $this->code = $code;
    }

}

File: src/app/src/main/java/com/example/issac/myapplication/MODEL/Specialisation.java
package com.example.issac.myapplication.MODEL;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;

public class Specialisation extends Model implements Serializable
{
    protected int id;
    private String libelle;
    private String code;

    public Specialisation() {
        super("specialisation");
    }

    public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public void putInObj(JSONObject json) {
        try {
            this.code = json.getString("code");                 // ici on assigne la totalité des attributs avec le résultat en JSON (ici modif pour debug req Creation)
            this.libelle = json.getString("libelle");
            this.id = json.getInt("id");

        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    /* Matthieu
    24/05/18
    Retourne un tableau contenant des objets specialisation
    Permet de convertir un tableau contenant des objets JSON en specialisations
     */
    public ArrayList<Specialisation> JSONArrayToSpecialisations(ArrayList<JSONObject> table) {
        ArrayList<Specialisation> lesSpecialisations = new ArrayList<Specialisation>();
        int i;
        for(i=0;i<table.size();i++) {
            Specialisation laSpecialisation = new Specialisation();
            laSpecialisation.putInObj(table.get(i));
            lesSpecialisations.add(laSpecialisation);
        }
        return lesSpecialisations;
    }

}
